app/services/UnloadingRemarksService.scala

```diff
@@ -16,78 +16,52 @@
 
 package services
 
-import java.time.LocalDate
-
+import cats.data.OptionT
 import com.google.inject.Inject
 import connectors.UnloadingConnector
 import logging.Logging
 import models.messages._
 import models.{ArrivalId, UnloadingPermission, UserAnswers}
 import pages._
-import play.api.http.Status._
 import repositories.InterchangeControlReferenceIdRepository
 import uk.gov.hmrc.http.HeaderCarrier
 
+import java.time.LocalDate
 import scala.concurrent.{ExecutionContext, Future}
 
-class UnloadingRemarksService @Inject() (metaService: MetaService,
-                                         remarksService: RemarksService,
-                                         unloadingRemarksRequestService: UnloadingRemarksRequestService,
-                                         interchangeControlReferenceIdRepository: InterchangeControlReferenceIdRepository,
-                                         unloadingRemarksMessageService: UnloadingRemarksMessageService,
-                                         unloadingConnector: UnloadingConnector
+class UnloadingRemarksService @Inject() (
+  metaService: MetaService,
+  remarksService: RemarksService,
+  unloadingRemarksRequestService: UnloadingRemarksRequestService,
+  interchangeControlReferenceIdRepository: InterchangeControlReferenceIdRepository,
+  unloadingRemarksMessageService: UnloadingRemarksMessageService,
+  unloadingConnector: UnloadingConnector
 )(implicit ec: ExecutionContext)
     extends Logging {
 
   def submit(arrivalId: ArrivalId, userAnswers: UserAnswers, unloadingPermission: UnloadingPermission)(implicit hc: HeaderCarrier): Future[Option[Int]] =
-    interchangeControlReferenceIdRepository
-      .nextInterchangeControlReferenceId()
-      .flatMap {
-        interchangeControlReference =>
-          remarksService
-            .build(userAnswers, unloadingPermission)
-            .flatMap {
-              unloadingRemarks =>
-                val meta: Meta = metaService.build(interchangeControlReference)
-
-                val unloadingRemarksRequest: UnloadingRemarksRequest =
-                  unloadingRemarksRequestService.build(meta, unloadingRemarks, unloadingPermission, userAnswers)
-
-                unloadingConnector
-                  .post(arrivalId, unloadingRemarksRequest)
-                  .flatMap(
-                    response => Future.successful(Some(response.status))
-                  )
-                  .recover {
-                    case ex =>
-                      logger.error(s"$ex")
-                      Some(SERVICE_UNAVAILABLE)
-                  }
```

Comment:
This submit method is only being used in the CheckYourAnswersController. A `None` and `Some(5xx)` both result in `errorHandler.onClientError(request, INTERNAL_SERVER_ERROR)` so I took the decision to remove this recover block